<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The hypogen package â€¢ hypogen</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="The hypogen package">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hypogen</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9001</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/hypogen_basics.html">The hypogen package</a>
    </li>
    <li>
      <a href="../articles/hypogen_genotype_frequncies.html">Genotype frequencies</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>The hypogen package</h1>
                        <h4 class="author">Kosmas Hench</h4>
            
            <h4 class="date">2019-03-03</h4>
      
      
      <div class="hidden name"><code>hypogen_basics.Rmd</code></div>

    </div>

    
    
<style type="text/css">
img {
  border: 0;
}
</style>
<div id="about-the-package" class="section level2">
<h2 class="hasAnchor">
<a href="#about-the-package" class="anchor"></a>About the package:</h2>
<p>The <strong>hypogen</strong> package helps with population genetic analysis based on the hamlet reference genome (<em>Hypoplectrus puella</em>). It contains two functions to import population genetic statistics as produced by e.g. <a href="https://vcftools.github.io/index.html">VCFtools</a> (<code><a href="../reference/hypo_import_snps.html">hypo_import_snps()</a></code> &amp; <code><a href="../reference/hypo_import_windows.html">hypo_import_windows()</a></code>) in the context of the hamlet genome. Furthermore it provides one template for genome wide plots (<code><a href="../reference/geom_hypo_LG.html">geom_hypo_LG()</a></code>) and one template to set statistics into the context of the genome annotation (<code><a href="../reference/hypo_annotation_baseplot.html">hypo_annotation_baseplot()</a></code>).</p>
<p>In addition to this the package contains a few helpers for a consistent plot layout. More <em>artistic</em> aspects of hamlet analysis plots are provided within the <a href="www.github.com/k-hench/hypoimg"><strong>hypoimg</strong></a> package.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Using the <strong>devtools</strong> package, <strong>hypogen</strong> can be installed directly from github:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"k-hench/hypogen"</span>)</code></pre></div>
</div>
<div id="importing-data" class="section level2">
<h2 class="hasAnchor">
<a href="#importing-data" class="anchor"></a>Importing Data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># SNP data</span>
file_snps &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"example.weir.fst.gz"</span>, <span class="dt">package =</span> <span class="st">"hypogen"</span>)
data_snps &lt;-<span class="st"> </span><span class="kw"><a href="../reference/hypo_import_snps.html">hypo_import_snps</a></span>(<span class="dt">file =</span> file_snps, <span class="dt">gz =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">window =</span> <span class="st">'bolditalic(F[ST])'</span>)

<span class="co"># Window data</span>
file_windows &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"example.windowed.weir.fst.gz"</span>, <span class="dt">package =</span> <span class="st">"hypogen"</span>)
data_windows &lt;-<span class="st"> </span><span class="kw"><a href="../reference/hypo_import_windows.html">hypo_import_windows</a></span>(<span class="dt">file =</span> file_windows, <span class="dt">gz =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">window =</span> <span class="st">'bolditalic(F[ST])'</span>)</code></pre></div>
</div>
<div id="the-basic-genome-wide-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#the-basic-genome-wide-plot" class="anchor"></a>The basic genome wide plot</h2>
<p>Ploting a certain statistic over the whole genome is a frequent issue. An example for this is the running average of <em>F<sub>ST</sub></em>.</p>
<p>Assume we have a dataset containing to variables measure along the genome:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">test_df &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">x=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq</a></span>(<span class="dv">1</span>,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Extremes">max</a></span>(hypo_karyotype<span class="op">$</span>GEND),<span class="dt">length.out =</span> <span class="dv">500</span>),<span class="dv">2</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CHROM =</span> <span class="kw"><a href="../reference/hypo_which_CHROM.html">hypo_which_CHROM</a></span>(x),<span class="dt">grp =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'sin'</span>,<span class="st">'cos'</span>), <span class="dt">each =</span> <span class="dv">500</span>),
         <span class="dt">y =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ifelse">ifelse</a></span>(grp<span class="op">==</span><span class="st">'sin'</span>,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Trig">sin</a></span>(x<span class="op">/</span><span class="dv">200000000</span><span class="op">*</span><span class="dv">2</span><span class="op">*</span>pi),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Trig">cos</a></span>(x<span class="op">/</span><span class="dv">200000000</span><span class="op">*</span><span class="dv">2</span><span class="op">*</span>pi<span class="op">+</span>.<span class="dv">2</span>))<span class="op">+</span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Normal">rnorm</a></span>(<span class="dv">1000</span>))
test_df <span class="op">%&gt;%</span><span class="st"> </span>head
<span class="co">#&gt; # A tibble: 6 x 4</span>
<span class="co">#&gt;          x CHROM grp        y</span>
<span class="co">#&gt;      &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt;</span>
<span class="co">#&gt; 1       1  LG01  sin   -0.529</span>
<span class="co">#&gt; 2 1121543. LG01  sin   -0.102</span>
<span class="co">#&gt; 3 2243086. LG01  sin    0.940</span>
<span class="co">#&gt; 4 3364628. LG01  sin   -1.09 </span>
<span class="co">#&gt; 5 4486171. LG01  sin   -1.02 </span>
<span class="co">#&gt; 6 5607713. LG01  sin    0.456</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="../reference/geom_hypo_LG.html">geom_hypo_LG</a></span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data=</span>test_df,<span class="kw">aes</span>(<span class="dt">x=</span>x,<span class="dt">y=</span>y,<span class="dt">col=</span>CHROM)) <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="../reference/scale_fill_hypo_LG_bg.html">scale_fill_hypo_LG_bg</a></span>() <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="../reference/scale_x_hypo_LG.html">scale_x_hypo_LG</a></span>() <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="../reference/scale_color_hypo_LG.html">scale_color_hypo_LG</a></span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">facet_grid</span>(grp<span class="op">~</span>.)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="../reference/theme_hypo.html">theme_hypo</a></span>()</code></pre></div>
<p><img src="hypogen_basics_files/figure-html/plot_genome_wide-1.png" width="100%"></p>
</div>
<div id="the-annotation-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#the-annotation-plot" class="anchor"></a>The annotation plot</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">XR &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">10000</span>,<span class="dv">50000</span>)
test_df_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">x=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq</a></span>(XR[<span class="dv">1</span>],XR[<span class="dv">2</span>],<span class="dt">length.out =</span> <span class="dv">200</span>),<span class="dv">2</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">window =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'bold(hyp_sin)'</span>,<span class="st">'bold(hyp_cos)'</span>), <span class="dt">each =</span> <span class="dv">200</span>),
         <span class="dt">y =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ifelse">ifelse</a></span>(window <span class="op">==</span><span class="st"> 'bold(hyp_sin)'</span>,
                    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Trig">sin</a></span>(x<span class="op">/</span>XR[<span class="dv">2</span>]<span class="op">*</span><span class="dv">20</span><span class="op">*</span>pi),
                    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Trig">cos</a></span>(x<span class="op">/</span>XR[<span class="dv">2</span>]<span class="op">*</span><span class="dv">20</span><span class="op">*</span>pi<span class="op">+</span>.<span class="dv">2</span>)) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Normal">rnorm</a></span>(<span class="dv">400</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/hypo_annotation_baseplot.html">hypo_annotation_baseplot</a></span>(<span class="dt">searchLG =</span> <span class="st">"LG01"</span>, <span class="dt">xrange =</span> XR,
                         <span class="dt">genes_of_interest =</span> <span class="st">"TSEN34"</span>,
                         <span class="dt">genes_of_sec_interest =</span> <span class="st">"Serp1"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> data_snps, <span class="kw">aes</span>(<span class="dt">x =</span> POS, <span class="dt">y =</span> WEIR_AND_COCKERHAM_FST),
            <span class="dt">color =</span> <span class="st">'lightgray'</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data =</span> data_windows, <span class="kw">aes</span>(<span class="dt">x =</span> POS, <span class="dt">y =</span> WEIGHTED_FST),
            <span class="dt">color =</span> viridis<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/viridis/topics/reexports">inferno</a></span>(<span class="dv">5</span>)[<span class="dv">4</span>]) <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim =</span> XR) <span class="op">+</span>
<span class="st">  </span><span class="kw">facet_grid</span>(window<span class="op">~</span>.,<span class="dt">scales=</span><span class="st">'free_y'</span>,
             <span class="dt">switch =</span> <span class="st">'y'</span>,<span class="dt">labeller =</span> label_parsed) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'black'</span>,<span class="st">'gray'</span>,viridis<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/viridis/topics/reexports">inferno</a></span>(<span class="dv">3</span>)),
                     <span class="dt">guide =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> viridis<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/viridis/topics/reexports">inferno</a></span>(<span class="dv">3</span>),
                    <span class="dt">guide =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">name =</span> <span class="st">"Hypo AnnoPlot Title"</span>,
                     <span class="dt">expand =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>,<span class="dv">0</span>),<span class="dt">position =</span> <span class="st">'top'</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="../reference/theme_hypo.html">theme_hypo</a></span>() <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="../reference/theme_hypo_anno_extra.html">theme_hypo_anno_extra</a></span>()
<span class="co">#&gt; Warning in .get_version_from_pragmas(pragmas): unrecognized GFF version</span>
<span class="co">#&gt; specified in the file</span></code></pre></div>
<p><img src="hypogen_basics_files/figure-html/plot_annotation-1.png" width="100%"></p>
<hr>
<center>
<img src="logo.svg" width="20%">
</center>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#about-the-package">About the package:</a></li>
      <li><a href="#installation">Installation</a></li>
      <li><a href="#importing-data">Importing Data</a></li>
      <li><a href="#the-basic-genome-wide-plot">The basic genome wide plot</a></li>
      <li><a href="#the-annotation-plot">The annotation plot</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Kosmas Hench.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
