<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genotype frequencies • hypogen</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genotype frequencies">
<meta property="og:description" content="">
<meta property="og:image" content="https://k-hench.github.io/hypogen/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hypogen</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/hypogen.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/hypogen_genotype_frequncies.html">Genotype frequencies</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/k-hench/hypogen">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Genotype frequencies</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/k-hench/hypogen/blob/master/vignettes/hypogen_genotype_frequncies.Rmd"><code>vignettes/hypogen_genotype_frequncies.Rmd</code></a></small>
      <div class="hidden name"><code>hypogen_genotype_frequncies.Rmd</code></div>

    </div>

    
    
<style type="text/css">
img {
  border: 0;
}
</style>
<div id="disclaimer" class="section level2">
<h2 class="hasAnchor">
<a href="#disclaimer" class="anchor"></a>Disclaimer:</h2>
<p>This used to work - unfortunately {ggtern} seems to be broken at tihs moment. This page will be updated if {ggtern} gets fixed…</p>
</div>
<div id="preparation" class="section level2">
<h2 class="hasAnchor">
<a href="#preparation" class="anchor"></a>Preparation:</h2>
<p>To be able to plot genotytpe frequencies, we need a genotype table where genotypes are encoded as 0,1, &amp; 2 (homozygous reference allele, heterozygous and homozygous alternative allele).</p>
<p>This can be easily generated from a <strong>vcf file</strong> using <strong>VCFtools</strong>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="ex">vcftools</span> \</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">  --gzvcf input.vcf.gz \</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">  --out  genotype_table \</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">  --012</a></code></pre></div>
<p>Unfortunately the <strong>vcftools</strong> output has the wrong orientation (samples as rows, sites as columns). So before we reed the genotypes into <strong>R</strong> they are transformed using <strong>awk</strong>:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="fu">awk</span> <span class="st">'</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="st">{</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="st">    for (i=1; i&lt;=NF; i++)  {</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="st">        a[NR,i] = $i</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="st">    }</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="st">}</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="st">NF&gt;p { p = NF }</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="st">END {</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="st">    for(j=1; j&lt;=p; j++) {</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="st">        str=a[1,j]</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="st">        for(i=2; i&lt;=NR; i++){</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="st">            str=str" "a[i,j];</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="st">        }</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="st">        print str</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="st">    }</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="st">}'</span> genotype_table.012 <span class="kw"><a href="https://rdrr.io/r/base/Logic.html">|</a></span> <span class="fu">gzip</span> <span class="op">&gt;</span> genotype_table.012.trans.txt.gz</a></code></pre></div>
</div>
<div id="r-setup" class="section level2">
<h2 class="hasAnchor">
<a href="#r-setup" class="anchor"></a>R setup</h2>
<p>To efficiently plot the genotypes frequencies we make use of the <strong>ggtern</strong> package as well as of the <strong>tidyverse</strong> (and of the <strong>hypogen</strong> package):</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(tidyverse)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ggtern)</a></code></pre></div>
</div>
<div id="loading-the-data-into-r" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-the-data-into-r" class="anchor"></a>Loading the data into R</h2>
<p>The package <strong>hypogen</strong> provides the function <code><a href="../reference/hypo_import_genotype_freq.html">hypo_import_genotype_freq()</a></code> which can read the transformed genotypee table and directly computes genotype frequencies:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">file_genotypes &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"genotype_table.012.trans.txt.gz"</span>, <span class="dt">package =</span> <span class="st">"hypogen"</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"></a>
<a class="sourceLine" id="cb4-3" data-line-number="3">genotype_freqs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/hypo_import_genotype_freq.html">hypo_import_genotype_freq</a></span>(file_genotypes)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; Rows: 89</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; Cols: 111</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt; dbl [111]: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,...</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; Call `spec()` for a copy-pastable column specification</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; Specify the column types with `col_types` to quiet this message</span></a></code></pre></div>
<p>This table can be directly plottet using <strong>gggtern</strong>. As a reference we also make use of the <strong>hypogen</strong> function <code><a href="../reference/hypo_hwe.html">hypo_hwe()</a></code> to indicate Hardy-Weinberg genotype frequencies:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">ggplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="st">  </span><span class="kw">coord_tern</span>()<span class="op">+</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> genotype_freqs,</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">             <span class="kw">aes</span>(<span class="dt">x =</span> AA, <span class="dt">y =</span> Aa, <span class="dt">z =</span> aa ),</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">             <span class="dt">col =</span> hypo_clr_LGs[<span class="dv">18</span>]) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_path</span>(<span class="dt">data =</span> <span class="kw"><a href="../reference/hypo_hwe.html">hypo_hwe</a></span>(<span class="dt">n =</span> <span class="dv">100</span>),</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">            <span class="kw">aes</span>(<span class="dt">x =</span> AA, <span class="dt">y =</span> Aa, <span class="dt">z =</span> aa ),</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">            <span class="dt">color =</span><span class="st">'black'</span>,<span class="dt">alpha =</span> <span class="fl">.6</span>)</a></code></pre></div>
<hr>
<center>
<img src="hypogen_genotype_frequncies_files/figure-html/unnamed-chunk-2-1.png" width="150pt" height="150pt">
</center>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#disclaimer">Disclaimer:</a></li>
      <li><a href="#preparation">Preparation:</a></li>
      <li><a href="#r-setup">R setup</a></li>
      <li><a href="#loading-the-data-into-r">Loading the data into R</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://k-hench.github.io/kh/">Kosmas Hench</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
